enhancements/support-cluster-HA-API.md

```diff
@@ -0,0 +1,131 @@
+
+```go
+
+// ClusterConfig contains the configuration information of the
+// ocp cluster.
+type ClusterConfig struct {
+    // ControlPlaneTopology contains the expectations for operands that 
+    // run on control plane nodes.
+    ControlPlaneTopology AvailabilityStatus
+
+    // InfrastructureTopology contains the expectations for infrastructre
+    // services.
+    InfrastructureTopology AvailabilityStatus
+}
+
+// Values which will be used to indicate the highly available
+// or single node deployment topology.
+const (
+    HighlyAvailable AvailabilityStatus = "HighlyAvailable"
+    SingleReplica AvailabilityStatus = "SingleReplica"
+)
+
+type AvailabilityStatus string
+
+```
+
+**Note**
+In future, the `ClusterConfig` can be extended to store additional information on cluster topology. 
+
+The ClusterConfig will contain the expectations for control plane and infrastructure nodes. Here:
+1. `ControlPlaneTopology` - refers to the expectations for operands that normally run on [control plane][control_plane] (or master) nodes.
+2. `InfrastructureTopology` - refers to the expectations for infrastructure services that do not run on control plane nodes, usually indicated by a node selector for a `role` value other than `master`.
+
+The possible values for both can be either `HighlyAvailable` or `SingleReplica`. More details on when the cluster is considered to support highly available or single replica deployments can be found [here][enhancement_cluster_operators].
+
+The library will provide an helper which will return the `ClusterConfig` mentioned above by reading the `infrastructure.config.openshift.io` crd present in cluster.
+
+```go
+// GetClusterConfig will return the HA expectations of the cluster
+func GetClusterConfig(kubeconfigPath string) ClusterConfig {
+    // Implementation will involve reading the required fields
+    // in status subreasource of `infrastructure.config.openshift.io`
+    // crd.
+}
+```
+
+**NOTE:**
+Operators cannot modify the informational resources present in OpenShift clusters. This library can only be used to retrieve information about the high availability expectations of the cluster and not to modify it.
+
+### Scenarios:
+
+1. Operator author tries to fetch the `ClusterConfig` for a non-openshift cluster:
+
+In this case, the library will return an error. The error can be wrapped, such that users can utilize the `errors.Is` method to identify this specific case.
+
+2. Operator author tries to use this library with older versions of ocp clusters which do not have HA expectations embedded in CRD:
+
+In this case, the error returned will be in the same format as stated in previous scenario but would specifically mention that the requested information cannot be found in CRD.
+
+### Restrictions of this library:
+
+1. Though the library is present in `operator-lib` (which has helpers for operator developers), it is not being directly used by `operator-sdk`:   
```

Comment:
I think library here would be `function` because `operator-lib` itself is a library. 
Also I'd probably rephrase this:

While `operator-sdk` will not scaffold this function in operators, the function is available to operator developers through `operator-lib`.